agent/app/agent.go

```diff
@@ -188,9 +191,18 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		seelog.Criticalf("Error creating state manager: %v", err)
 		return exitcodes.ExitTerminal
 	}
-	// TODO:
-	// 1) Feature gate ECS cgroup root init
-	// 2) Update mocks and unit tests
+
+	// Conditionally create '/ecs' cgroup root
+	if agent.cfg.TaskCPUMemLimit {
+		err = agent.resource.Init()
+		// When task CPU and memory limits are enabled, all tasks are placed
+		// under the '/ecs' cgroup root. If we are unable to create this
+		// cgroup, the agent should exit with a terminal error.
```

Comment:
This should *only* be fatal if task resource limits are *explicitly* enabled (rather than just implicitly enabled).  The reason is that a customer may call `UpdateContainerAgent` on an instance that is running an old version of `ecs-init` that doesn't mount in the cgroup hierarchy; we should not fail to restart the agent simply because a feature that they're not using doesn't work.

Reply:
This is a really good point but this call is conditioned on the `config`. 
We are going to have another PR to conditionally disable the feature and we can ensure that the cgroup mounts are valid at that point.

Reply:
Can you add a TODO if you're not going to address this now?  We need to get this done correctly before release, but it doesn't need to be part of this PR.

Reply:
I will add a `TODO` here in my next revision
It will be addressed in a different PR though

Reply:
`TODO` has been added above on `L196`. I'll refine the TODO's here